java/core/src/java/org/apache/orc/geospatial/BoundingBox.java

```diff
@@ -0,0 +1,374 @@
+   * and the extent of the other bounding box.
+   *
+   * @param other the other BoundingBox whose bounds will be merged into this one
+   */
+  public void merge(BoundingBox other) {
+    if (!valid) {
+      return;
+    }
+
+    // If other is null or invalid, mark this as invalid
+    if (other == null || !other.valid) {
+      valid = false;
+      resetBBox();
+      return;
+    }
+
+    this.xMin = Math.min(this.xMin, other.xMin);
+    this.xMax = Math.max(this.xMax, other.xMax);
+    this.yMin = Math.min(this.yMin, other.yMin);
+    this.yMax = Math.max(this.yMax, other.yMax);
+    this.zMin = Math.min(this.zMin, other.zMin);
+    this.zMax = Math.max(this.zMax, other.zMax);
+    this.mMin = Math.min(this.mMin, other.mMin);
+    this.mMax = Math.max(this.mMax, other.mMax);
+
+    // Update the validity of this bounding box based on the other bounding box
+    valid = isXYValid();
+  }
+
+  /**
+   * Extends this bounding box to include the spatial extent of the provided geometry.
+   * The bounding box coordinates (min/max values for x, y, z, m) will be adjusted
+   * to encompass both the current bounds and the geometry's bounds.
+   *
+   * @param geometry The geometry whose coordinates will be used to update this bounding box.
+   *                 If null or empty, the method returns without making any changes.
+   */
+  public void update(Geometry geometry) {
+    if (!valid) {
+      return;
+    }
+
+    if (geometry == null || geometry.isEmpty()) {
+      return;
+    }
+
+    Envelope envelope = geometry.getEnvelopeInternal();
+    updateBounds(envelope.getMinX(), envelope.getMaxX(), envelope.getMinY(), envelope.getMaxY());
+
+    for (Coordinate coord : geometry.getCoordinates()) {
+      if (!Double.isNaN(coord.getZ())) {
+        zMin = Math.min(zMin, coord.getZ());
+        zMax = Math.max(zMax, coord.getZ());
+      }
+      if (!Double.isNaN(coord.getM())) {
+        mMin = Math.min(mMin, coord.getM());
+        mMax = Math.max(mMax, coord.getM());
+      }
+    }
+
+    // Update the validity of this bounding box based on the other bounding box
+    valid = isXYValid();
+  }
+
+  /**
+   * Updates the X and Y bounds of this bounding box with the given coordinates.
+   * Updates are conditional:
+   * - X bounds are only updated if both minX and maxX are not NaN
+   * - Y bounds are only updated if both minY and maxY are not NaN
+   * This allows partial updates while preserving valid dimensions.
+   */
+  private void updateBounds(double minX, double maxX, double minY, double maxY) {
+    if (!Double.isNaN(minX) && !Double.isNaN(maxX)) {
+      xMin = Math.min(xMin, minX);
+      xMax = Math.max(xMax, maxX);
+    }
+
+    if (!Double.isNaN(minY) && !Double.isNaN(maxY)) {
+      yMin = Math.min(yMin, minY);
+      yMax = Math.max(yMax, maxY);
+    }
+  }
+
+  /**
+   * Aborts the bounding box by resetting it to its initial state.
+   */
+  public void abort() {
+    valid = false;
+    resetBBox();
```

Comment:
Ditto. We need to switch the order of the statements if we need to keep `valid=false`.